### DISCUSSION TITLE
Understand the signerOrProvider optional configuration.

### DISCUSSION THREAD

**Discussion by salieflewis** [wevm/wagmi#1794] [Q&A]
I've instantiated a contract instance using `useContract` and then later I'm performing a read call with vanilla ethers. Basically one of the read call's did not go through until I applied the signerOrProvider config to the respective `useContract` instance, and then passed the object returned from `useProvider` 

Basically I don't exactly understand why that change would make or break whether the call was successful, and was hoping someone could share some light on that in simplified terms.

**Comment by tmm** [comment]
Hard to tell what is going on in this case without a minimal reproducible example.

**Reply by salieflewis** [reply]
Contract instantiation is here 
```
const tokenContract = useContract({
    address: tokenAddress,
    abi: tokenAbi,
    signerOrProvider: provider,
  })
```

Call is here
```
 React.useEffect(() => {
    async function getSupply() {
      try {
        const supply = await tokenContract?.totalSupply()
        setTotalSupply(supply?.toNumber())
      } catch (err) {
        console.error(err)
      }
    }
    getSupply()
  }, [tokenContract, auctionData?.tokenId])
```
Compatible `tokenAddress` is 0xdf9b7d26c8fc806b1ae6273684556761ff02d422

--

I hope that provides more clarity. Not sure what other factors might be at play that would introduce more context.
